compose/ui/ui-text/src/desktopMain/kotlin/androidx/compose/ui/text/platform/AwtFontInterop.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.text.platform
+
+import androidx.compose.ui.text.ExperimentalTextApi
+import androidx.compose.ui.text.font.EmbeddedFontFamily
+import androidx.compose.ui.text.font.FontFamily
+import java.awt.Font
+import org.jetbrains.skiko.OS
+import org.jetbrains.skiko.hostOs
+
+/**
+ * Obtain a [FontFamily] corresponding to this AWT [Font]. The Compose
+ * runtime is able to match font families available in the system, and
+ * ones embedded in the JetBrains Runtime (see note below). If the [Font]
+ * is a logical font (i.e., one of [Font.DIALOG], [Font.DIALOG_INPUT],
+ * [Font.SERIF], [Font.SANS_SERIF], or [Font.MONOSPACED]) then the actual
+ * physical font will be resolved.
+ *
+ * If this [Font] is not known to Compose — e.g., because it's been created
+ * from a file — the lookup will return a fallback font family. If you need
+ * to create a font family from font files instead, you can do it like
+ * this:
+ * ```kotlin
+ * val myFontFamily = FontFamily(
+ *     File(myRegularFontFile, FontWeight.Normal, FontStyle.Normal),
+ *     File(myItalicFontFile, FontWeight.Normal, FontStyle.Italic),
+ *     // ...
+ * )
+ * ```
+ *
+ * **Note:** this will return a fallback font family, such as
+ * [FontFamily.Default] or [FontFamily.Monospace], when running on Java
+ * VMs other than the JetBrains Runtime, and/or if the current module
+ * cannot access the `sun.font` module. To ensure the `sun.font` module
+ * is accessible, you should pass this argument to the JVM: `--add-opens
+ * java.desktop/sun.font=ALL-UNNAMED`.
+ *
+ * **Note:** on macOS, calling this on a reserved system font, such
+ * as `.AppleSystemUI*` families, will return [FontFamily.Default].
+ * Reserved system fonts aren't supposed to be accessed directly from
+ * apps, and forcing it will not work anyway, as Skia doesn't support it.
+ *
+ * @see resolveFontFamilyNameOrNull
+ * @see fontFamilyNameOrNull
+ */
+@ExperimentalTextApi
+fun Font.asFontFamily(): FontFamily {
```

Comment:
```suggestion
fun Font.asComposeFontFamily(): FontFamily {
```

Compose have this naming convention (for examaple, we have `asComposeImageBitmap`